packages/components/pie-modal/src/index.ts

```diff
@@ -108,21 +108,36 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
 
     private _backButtonClicked = false;
 
+    private _eventsListenerController : AbortController | null = null;
+
     // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
 
     connectedCallback () : void {
         super.connectedCallback();
-        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event));
-        document.addEventListener(ON_MODAL_OPEN_EVENT, (event) => this._handleModalOpened(<CustomEvent>event));
-        document.addEventListener(ON_MODAL_CLOSE_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
-        document.addEventListener(ON_MODAL_BACK_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
+        this._eventsListenerController = new AbortController();
+
+        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event), {
+            signal: this._eventsListenerController.signal,
+        });
```

Comment:
Just a formatting question for if we do decide to implement this, could we destructure `signal` first?

```suggestion
        this._eventsListenerController = new AbortController();
        const { signal } = this._eventsListenerController;

        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event), { signal });
```

Reply:
can do!